crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -187,6 +189,11 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
         None => None,
     };
 
+    let global_reporter = match args.format {
```

Comment:
Why was `global_reporter` added?  This requires knowledge of the reporter's implementation to know whether it should be global or not which makes this more brittle

Reply:
Because all previous reporter require only partial state, the output can be written as soon as any event arrives. But sarif must wait until all processing is complete, then collect the output and aggregate it into a single file.

Reply:
Maybe put another way, why can't they all be global?

Reply:
I think it's possible, but it would require staging all events while processing, adding an unnecessary memory footprint. I can refactor the old reporter if you think simplicity in code is more important.

Reply:
Not seeing why we'd need to do that.  I just pushed my suggested change and it seems to work.